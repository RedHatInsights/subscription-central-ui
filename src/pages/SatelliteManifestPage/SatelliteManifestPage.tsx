import React, { FC } from 'react';
import PageHeader from '@redhat-cloud-services/frontend-components/PageHeader';
import { Text, TextContent } from '@patternfly/react-core';
import SatelliteManifestPanel from '../../components/SatelliteManifestPanel';
import useSatelliteManifests from '../../hooks/useSatelliteManifests';
import Unavailable from '@redhat-cloud-services/frontend-components/Unavailable';
import { Processing } from '../../components/emptyState';
import useUser from '../../hooks/useUser';
import ExternalLink from '../../components/ExternalLink';
import { NoSatelliteSubs } from '../../components/NoSatelliteSubs';
import { Alert } from '@patternfly/react-core';
import { subscriptionInventoryLink, supportLink } from '../../utilities/consts';
import { PageSection } from '@patternfly/react-core';
import { useNavigate } from 'react-router-dom';

const SatelliteManifestPage: FC = () => {
  const { isLoading, isFetching, error, data } = useSatelliteManifests();
  const navigate = useNavigate();

  const { data: user } = useUser();
  const manifestsMoreInfoLink =
    'https://access.redhat.com/documentation/en-us/subscription_central/2023/html/' +
    'creating_and_managing_manifests_for_a_connected_satellite_server/index';

  if (!user.canReadManifests) {
    navigate('./no-permissions');
  }

  return (
    <>
      <PageHeader>
        <TextContent>
          <Text component="h1">Manifests</Text>
          <Text component="p">
            Create and export manifests for your on-premise subscription management application.
            This page does not allow you to add subscriptions to manifests. Subscriptions can be
            added to the manifest using using the Satellite Server web user interface after
            importing the downloaded manifests.{' '}
            <ExternalLink href={manifestsMoreInfoLink}>
              Learn more about creating and managing manifests for a connected Satellite Server
            </ExternalLink>
            {!user.isEntitled && !isLoading && data?.length != 0 && (
              <Alert title="Your account has no Satellite subscriptions" isInline variant="info">
                You can view existing manifests for your account, however, an active Satellite
                subscription is required to create a new manifest.{' '}
                <a href={supportLink} target="_blank" rel="noreferrer">
                  Contact support
                </a>{' '}
                to determine if you need a new subscription. To view recently expired subscriptions,
                select the <em>Expired</em> card in your{' '}
                <a href={subscriptionInventoryLink}>subscription inventory</a>.
              </Alert>
            )}
          </Text>
        </TextContent>
      </PageHeader>
      <PageSection>
        <>
          {isLoading && !error && <Processing />}

          {!isLoading && !error && !user.isEntitled && data.length == 0 && <NoSatelliteSubs />}

          {!isLoading && !error && (user.isEntitled || data.length != 0) && (
            <SatelliteManifestPanel data={data} user={user} isFetching={isFetching} />
          )}

          {error && <Unavailable />}
        </>
      </PageSection>
    </>
  );
};

export default SatelliteManifestPage;
