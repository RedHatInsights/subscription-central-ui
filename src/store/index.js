import { getRegistry } from '@redhat-cloud-services/frontend-components-utilities/Registry';
import promiseMiddleware from 'redux-promise-middleware';

let registry;

export function init(...middleware) {
  registry = getRegistry({}, [promiseMiddleware, ...middleware]);
  return registry;
}

export function getStore() {
  return registry.getStore();
}

export function register(...args) {
  return registry.register(...args);
}
