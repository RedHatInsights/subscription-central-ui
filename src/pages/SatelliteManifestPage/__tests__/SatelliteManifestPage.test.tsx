import React from 'react';
import { render } from '@testing-library/react';
import SatelliteManifestPage from '../SatelliteManifestPage';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { init } from '../../../store';
import { QueryClient, QueryClientProvider } from 'react-query';
import useSatelliteManifests from '../../../hooks/useSatelliteManifests';

jest.mock('../../../hooks/useSatelliteManifests');

const queryClient = new QueryClient();

describe('Satellite Manifests Page', () => {
  it('renders correctly with satellite data', () => {
    window.insights = {};
    (useSatelliteManifests as jest.Mock).mockReturnValue({
      isLoading: false,
      data: [
        {
          name: 'Sputnik',
          type: 'Satellite',
          url: 'www.example.com',
          uuid: '00000000-0000-0000-0000-000000000000',
          version: '1.2.3'
        }
      ]
    });

    const { container } = render(
      <Provider store={init().getStore()}>
        <Router>
          <QueryClientProvider client={queryClient}>
            <SatelliteManifestPage />
          </QueryClientProvider>
        </Router>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders the empty state with create manifest button when no results are returned', () => {
    window.insights = {};
    (useSatelliteManifests as jest.Mock).mockReturnValue({
      isLoading: false,
      data: []
    });

    const { container } = render(
      <Provider store={init().getStore()}>
        <Router>
          <QueryClientProvider client={queryClient}>
            <SatelliteManifestPage />
          </QueryClientProvider>
        </Router>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with an error message when an API fails', () => {
    window.insights = {};
    (useSatelliteManifests as jest.Mock).mockReturnValue({
      isLoading: false,
      error: true,
      data: undefined
    });

    const { container } = render(
      <Provider store={init().getStore()}>
        <Router>
          <QueryClientProvider client={queryClient}>
            <SatelliteManifestPage />
          </QueryClientProvider>
        </Router>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
