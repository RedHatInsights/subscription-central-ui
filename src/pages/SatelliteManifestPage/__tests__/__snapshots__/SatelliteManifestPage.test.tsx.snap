// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Satellite Manifests Page renders correctly with satellite data 1`] = `
<div>
  <section
    class="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
    widget-type="InsightsPageHeader"
  >
    <h1
      class="pf-c-title pf-m-2xl"
      widget-type="InsightsPageHeaderTitle"
    >
       
      Satellite Manifests
       
    </h1>
    <p>
      Export subscriptions to your on-premise subscription management application
    </p>
  </section>
  <section
    class="pf-l-page__main-section pf-c-page__main-section"
    page-type=""
  >
    <section
      class="pf-c-page__main-section pf-m-light"
    >
      <h2
        class="pf-c-title pf-m-xl"
      >
        Satellite Manifests
        <span
          class="pf-c-badge pf-m-read"
        >
          1
        </span>
      </h2>
      <div
        class="pf-l-flex pf-m-column pf-m-row-on-md pf-m-justify-content-space-between"
      >
        <div
          class=""
        >
          <div
            class="pf-l-split pf-m-gutter"
          >
            <div
              class="pf-l-split__item pf-m-fill"
            >
              <div
                class="pf-c-search-input"
              >
                <div
                  class="pf-c-input-group"
                >
                  <div
                    class="pf-c-search-input__bar"
                  >
                    <span
                      class="pf-c-search-input__text"
                    >
                      <span
                        class="pf-c-search-input__icon"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                          />
                        </svg>
                      </span>
                      <input
                        aria-label="Search input"
                        class="pf-c-search-input__text-input"
                        placeholder="Filter by name, version or UUID"
                        value=""
                      />
                    </span>
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              class="pf-l-split__item"
            >
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-primary"
                data-ouia-component-id="OUIA-Generated-Button-primary-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                Create new manifest
              </button>
            </div>
          </div>
        </div>
        <div
          class="pf-m-align-right"
        >
          <div
            class="pf-c-pagination"
            data-ouia-component-id="OUIA-Generated-Pagination-top-1"
            data-ouia-component-type="PF4/Pagination"
            data-ouia-safe="true"
            id="pagination-options-menu-0"
            style="--pf-c-pagination__nav-page-select--c-form-control--width-chars: 2;"
          >
            <div
              class="pf-c-pagination__total-items"
            >
              <b>
                1
                 - 
                1
              </b>
               
              of 
              <b>
                1
              </b>
               
              
            </div>
            <div
              class="pf-c-options-menu"
              data-ouia-component-type="PF4/PaginationOptionsMenu"
              data-ouia-safe="true"
            >
              <div
                class="pf-c-options-menu__toggle pf-m-plain pf-m-text"
              >
                <span
                  class="pf-c-options-menu__toggle-text"
                >
                  <b>
                    1
                     - 
                    1
                  </b>
                   
                  of 
                  <b>
                    1
                  </b>
                   
                  
                </span>
                <button
                  aria-expanded="false"
                  aria-label="Items per page"
                  class="  pf-c-options-menu__toggle-button"
                  data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                  data-ouia-component-type="PF4/DropdownToggle"
                  data-ouia-safe="true"
                  id="pagination-options-menu-toggle-0"
                  type="button"
                >
                  <span
                    class="pf-c-options-menu__toggle-button-icon"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <nav
              aria-label="Pagination"
              class="pf-c-pagination__nav"
            >
              <div
                class="pf-c-pagination__nav-control pf-m-first"
              >
                <button
                  aria-disabled="true"
                  aria-label="Go to first page"
                  class="pf-c-button pf-m-plain pf-m-disabled"
                  data-action="first"
                  data-ouia-component-id="OUIA-Generated-Button-plain-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  disabled=""
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 448 512"
                    width="1em"
                  >
                    <path
                      d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="pf-c-pagination__nav-control"
              >
                <button
                  aria-disabled="true"
                  aria-label="Go to previous page"
                  class="pf-c-button pf-m-plain pf-m-disabled"
                  data-action="previous"
                  data-ouia-component-id="OUIA-Generated-Button-plain-2"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  disabled=""
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="pf-c-pagination__nav-page-select"
              >
                <input
                  aria-label="Current page"
                  class="pf-c-form-control"
                  disabled=""
                  max="1"
                  min="1"
                  type="number"
                  value="1"
                />
                <span
                  aria-hidden="true"
                >
                  of 
                  1
                </span>
              </div>
              <div
                class="pf-c-pagination__nav-control"
              >
                <button
                  aria-disabled="true"
                  aria-label="Go to next page"
                  class="pf-c-button pf-m-plain pf-m-disabled"
                  data-action="next"
                  data-ouia-component-id="OUIA-Generated-Button-plain-3"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  disabled=""
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="pf-c-pagination__nav-control pf-m-last"
              >
                <button
                  aria-disabled="true"
                  aria-label="Go to last page"
                  class="pf-c-button pf-m-plain pf-m-disabled"
                  data-action="last"
                  data-ouia-component-id="OUIA-Generated-Button-plain-4"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  disabled=""
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 448 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                    />
                  </svg>
                </button>
              </div>
            </nav>
          </div>
        </div>
      </div>
      <table
        aria-label="Satellite Manifest Table"
        class="pf-c-table pf-m-grid-md"
        data-ouia-component-id="OUIA-Generated-Table-2"
        data-ouia-component-type="PF4/Table"
        data-ouia-safe="true"
        role="grid"
      >
        <thead
          class=""
        >
          <tr
            class=""
            data-ouia-component-id="OUIA-Generated-TableRow-1"
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe="true"
          >
            <th
              aria-sort="ascending"
              class="pf-c-table__sort pf-m-selected"
              data-key="0"
              data-label="Name"
              scope="col"
            >
              <button
                class="pf-c-table__button"
                type="button"
              >
                <div
                  class="pf-c-table__button-content"
                >
                  <span
                    class="pf-c-table__text"
                  >
                    Name
                  </span>
                  <span
                    class="pf-c-table__sort-indicator"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M88 166.059V468c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12V166.059h46.059c21.382 0 32.09-25.851 16.971-40.971l-86.059-86.059c-9.373-9.373-24.569-9.373-33.941 0l-86.059 86.059c-15.119 15.119-4.411 40.971 16.971 40.971H88z"
                      />
                    </svg>
                  </span>
                </div>
              </button>
            </th>
            <th
              aria-sort="none"
              class="pf-c-table__sort"
              data-key="1"
              data-label="Version"
              scope="col"
            >
              <button
                class="pf-c-table__button"
                type="button"
              >
                <div
                  class="pf-c-table__button-content"
                >
                  <span
                    class="pf-c-table__text"
                  >
                    Version
                  </span>
                  <span
                    class="pf-c-table__sort-indicator"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z"
                      />
                    </svg>
                  </span>
                </div>
              </button>
            </th>
            <th
              aria-sort="none"
              class="pf-c-table__sort pf-m-width-20"
              data-key="2"
              data-label="column-2"
              scope="col"
            >
              <button
                class="pf-c-table__button"
                type="button"
              >
                <div
                  class="pf-c-table__button-content"
                >
                  <span
                    class="pf-c-table__text"
                  >
                    Simple Content Access
                    <svg
                      aria-hidden="true"
                      class="sca-more-info-icon"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                      />
                    </svg>
                  </span>
                  <span
                    class="pf-c-table__sort-indicator"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z"
                      />
                    </svg>
                  </span>
                </div>
              </button>
            </th>
            <th
              aria-sort="none"
              class="pf-c-table__sort"
              data-key="3"
              data-label="UUID"
              scope="col"
            >
              <button
                class="pf-c-table__button"
                type="button"
              >
                <div
                  class="pf-c-table__button-content"
                >
                  <span
                    class="pf-c-table__text"
                  >
                    UUID
                  </span>
                  <span
                    class="pf-c-table__sort-indicator"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z"
                      />
                    </svg>
                  </span>
                </div>
              </button>
            </th>
          </tr>
        </thead>
        <tbody
          class=""
          role="rowgroup"
        >
          <tr
            class=""
            data-ouia-component-id="OUIA-Generated-TableRow-2"
            data-ouia-component-type="PF4/TableRow"
            data-ouia-safe="true"
          >
            <td
              class=""
              data-key="0"
              data-label="Name"
            >
              Sputnik
            </td>
            <td
              class=""
              data-key="1"
              data-label="Version"
            >
              1.2.3
            </td>
            <td
              class=""
              data-key="2"
              data-label="column-2"
            />
            <td
              class=""
              data-key="3"
              data-label="UUID"
            >
              00000000-0000-0000-0000-000000000000
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="pf-c-pagination pf-m-bottom"
        data-ouia-component-id="OUIA-Generated-Pagination-bottom-1"
        data-ouia-component-type="PF4/Pagination"
        data-ouia-safe="true"
        id="pagination-options-menu-1"
        style="--pf-c-pagination__nav-page-select--c-form-control--width-chars: 2;"
      >
        <div
          class="pf-c-options-menu pf-m-top"
          data-ouia-component-type="PF4/PaginationOptionsMenu"
          data-ouia-safe="true"
        >
          <div
            class="pf-c-options-menu__toggle pf-m-plain pf-m-text"
          >
            <span
              class="pf-c-options-menu__toggle-text"
            >
              <b>
                1
                 - 
                1
              </b>
               
              of 
              <b>
                1
              </b>
               
              
            </span>
            <button
              aria-expanded="false"
              aria-label="Items per page"
              class="  pf-c-options-menu__toggle-button"
              data-ouia-component-id="OUIA-Generated-DropdownToggle-2"
              data-ouia-component-type="PF4/DropdownToggle"
              data-ouia-safe="true"
              id="pagination-options-menu-toggle-1"
              type="button"
            >
              <span
                class="pf-c-options-menu__toggle-button-icon"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 320 512"
                  width="1em"
                >
                  <path
                    d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <nav
          aria-label="Pagination"
          class="pf-c-pagination__nav"
        >
          <div
            class="pf-c-pagination__nav-control pf-m-first"
          >
            <button
              aria-disabled="true"
              aria-label="Go to first page"
              class="pf-c-button pf-m-plain pf-m-disabled"
              data-action="first"
              data-ouia-component-id="OUIA-Generated-Button-plain-5"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              disabled=""
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 448 512"
                width="1em"
              >
                <path
                  d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                />
              </svg>
            </button>
          </div>
          <div
            class="pf-c-pagination__nav-control"
          >
            <button
              aria-disabled="true"
              aria-label="Go to previous page"
              class="pf-c-button pf-m-plain pf-m-disabled"
              data-action="previous"
              data-ouia-component-id="OUIA-Generated-Button-plain-6"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              disabled=""
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 256 512"
                width="1em"
              >
                <path
                  d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
                />
              </svg>
            </button>
          </div>
          <div
            class="pf-c-pagination__nav-page-select"
          >
            <input
              aria-label="Current page"
              class="pf-c-form-control"
              disabled=""
              max="1"
              min="1"
              type="number"
              value="1"
            />
            <span
              aria-hidden="true"
            >
              of 
              1
            </span>
          </div>
          <div
            class="pf-c-pagination__nav-control"
          >
            <button
              aria-disabled="true"
              aria-label="Go to next page"
              class="pf-c-button pf-m-plain pf-m-disabled"
              data-action="next"
              data-ouia-component-id="OUIA-Generated-Button-plain-7"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              disabled=""
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 256 512"
                width="1em"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                />
              </svg>
            </button>
          </div>
          <div
            class="pf-c-pagination__nav-control pf-m-last"
          >
            <button
              aria-disabled="true"
              aria-label="Go to last page"
              class="pf-c-button pf-m-plain pf-m-disabled"
              data-action="last"
              data-ouia-component-id="OUIA-Generated-Button-plain-8"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              disabled=""
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 448 512"
                width="1em"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                />
              </svg>
            </button>
          </div>
        </nav>
      </div>
    </section>
  </section>
</div>
`;

exports[`Satellite Manifests Page renders the empty state with create manifest button when no results are returned 1`] = `
<div>
  <section
    class="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
    widget-type="InsightsPageHeader"
  >
    <h1
      class="pf-c-title pf-m-2xl"
      widget-type="InsightsPageHeaderTitle"
    >
       
      Satellite Manifests
       
    </h1>
    <p>
      Export subscriptions to your on-premise subscription management application
    </p>
  </section>
  <section
    class="pf-l-page__main-section pf-c-page__main-section"
    page-type=""
  >
    <div
      class="pf-c-empty-state pf-m-lg"
    >
      <div
        class="pf-c-empty-state__content"
      >
        <svg
          aria-hidden="true"
          class="pf-c-empty-state__icon"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
          />
        </svg>
        <h5
          class="pf-c-title pf-m-lg"
        >
          Create a new manifest to export subscriptions
        </h5>
        <div
          class="pf-c-empty-state__body"
        >
          You currently have no manifests displayed. Create a new manifest to export subscriptions from the Red Hat Customer Portal to your on-premise subscription management application.
        </div>
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-primary"
          data-ouia-component-id="OUIA-Generated-Button-primary-2"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          Create new manifest
        </button>
      </div>
    </div>
  </section>
</div>
`;

exports[`Satellite Manifests Page renders with an error message when an API fails 1`] = `
<div>
  <section
    class="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
    widget-type="InsightsPageHeader"
  >
    <h1
      class="pf-c-title pf-m-2xl"
      widget-type="InsightsPageHeaderTitle"
    >
       
      Satellite Manifests
       
    </h1>
    <p>
      Export subscriptions to your on-premise subscription management application
    </p>
  </section>
  <section
    class="pf-l-page__main-section pf-c-page__main-section"
    page-type=""
  >
    <div
      class="pf-c-empty-state pf-m-lg ins-c-empty-state__unavailable pf-m-redhat-font"
    >
      <div
        class="pf-c-empty-state__content"
      >
        <svg
          aria-hidden="true"
          class="pf-c-empty-state__icon"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          />
        </svg>
        <h5
          class="pf-c-title pf-m-lg"
        >
          This page is temporarily unavailable
        </h5>
        <div
          class="pf-c-empty-state__body"
        >
          Try refreshing the page. If the problem persists, contact your organization administrator or visit our
          <a
            href="https://status.redhat.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
             status page
          </a>
           for known outages.
        </div>
      </div>
    </div>
  </section>
</div>
`;
